### PR TITLE
nbwsc's article

### PR DESCRIPTION
**Pull request by nbwsc**
Fixes # .

Make sure these boxes are checked before creating your pull request -- thank you!

- [x] Add Blog Header, including title, author, date.
e.g.
```
---
 layout: post
 title: "'Score Your Face Photo' a ML&Wechaty practice"
 date: 2017-09-18 09:00 +0800
 author: huyingxi
 ---
 
 > Author: [@huyingxi](https://github.com/huyingxi/wechaty_selfie) enjoying ML&Wechaty at BUPT
```
- [x] Keep all filenames & url as lowercase, and use `-` to connect words instead of space. e.g. `2017-10-06-wechat-pc-impactor` instead of `2017-10-06-WeChat PC Impactor`,
- [x] Embed the photo & video before publishing, save all external file to the blog `/download/2018` directory.
- [x] Add `<!--more-->` section to just show the abstract in blog homepage.

[See more](https://github.com/Chatie/blog#contribute-guideline)

### CODE REVIEW

File: _posts/2018-05-06-talk-about-wechat-robot.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+* [youfou/wxpy](https://github。com/youfou/wxpy) | 优秀的 api 包装和配套插件，微信机器人/优雅的微信个人号 API|[个人评价: 支持 python2/3 api 包装不错 文档写的满分 基于 itchat 开发，也开放了 itchat 的原始数据接口，自由度和封装美化兼顾 ]
+
+* [liuwons/wxBot](https://github。com/liuwons/wxBot)| 类似的基于 Python 的微信机器人 | [个人评价: 不支持 python3 utf-8 支持不好 ]
+
+* [sjdy521/Mojo-Weixin](https://github。com/sjdy521/Mojo-Weixin)|使用 Perl 语言编写的微信客户端框架，可通过插件提供基于 HTTP 协议的 api 接口供其他语言调用 |[未使用]
+
+* [HanSon/vbot](https://github。com/hanson/vbot)|基于 PHP7 的微信个人号机器人，通过实现匿名函数可以方便地实现各种自定义的功能 |[未使用]
+
+* [littlecodersh/ItChat](https://github。com/littlecodersh/ItChat)|wxpy 就是基于此开发
```
Comment (huan) [inline _posts/2018-05-06-talk-about-wechat-robot.md]: github。com -> github.com

File: _posts/2018-05-06-talk-about-wechat-robot.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+这种方式我本人没有尝试过， 我的同事用 C#实现了一个版本， pc 版的微信的界面也是 html 渲染的， 所以 windows 无法获得页面里面控件的句柄， 所以需要一些很 hack 的方式去做， 我不太喜欢这种(主要我不太熟 windows)。 这种的好处就是， 就算网页版被封了， pc 版暂时还不受什么影响。
+
+### 在 Android 下使用 hook 操作微信安卓版
+
+今年 4 月份的时候，运营提出需求需要群管理，几百个 100 人的群，需要自动回复，自动踢人(发重复消息的， 发关键字的， 发太长信息的。。。)，管理员踢人，防止该群名，黑名单(加入黑名单在所有群里都一进群就踢)等功能， 一开始我用`wechaty`做了一个版本， 但是第二天就被封号了(运气实在太差了喂)。 然后就想到 android 上面有 xposed 这个东西， 可以作为 android 的 hook 来修改一些 apk 的行为，摘一段官方的介绍:
+
+> Xposed is a framework for modules that can change the behavior of the system and apps without touching any APKs
+
+接下去就是找找现成模块有没有能用的， 如果不够用， 就找找有没有开源， 改改代码。 实在不行也可以自己写， xposed 模块还是挺好写的。对了还有一个叫[VirtualXposed](https://github。com/android-hacker/VirtualXposed)的东西，修改微信 hook 的事情，这个足够了， 不需要 root。
```
Comment (huan) [inline _posts/2018-05-06-talk-about-wechat-robot.md]: github。com -> github.com

File: _posts/2018-05-06-talk-about-wechat-robot.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+这种库就很多了，我们的`wechaty`核心也是这种方式，还有其他各种语言的实现:
+
+* [youfou/wxpy](https://github。com/youfou/wxpy) | 优秀的 api 包装和配套插件，微信机器人/优雅的微信个人号 API|[个人评价: 支持 python2/3 api 包装不错 文档写的满分 基于 itchat 开发，也开放了 itchat 的原始数据接口，自由度和封装美化兼顾 ]
+
+* [liuwons/wxBot](https://github。com/liuwons/wxBot)| 类似的基于 Python 的微信机器人 | [个人评价: 不支持 python3 utf-8 支持不好 ]
+
+* [sjdy521/Mojo-Weixin](https://github。com/sjdy521/Mojo-Weixin)|使用 Perl 语言编写的微信客户端框架，可通过插件提供基于 HTTP 协议的 api 接口供其他语言调用 |[未使用]
+
+* [HanSon/vbot](https://github。com/hanson/vbot)|基于 PHP7 的微信个人号机器人，通过实现匿名函数可以方便地实现各种自定义的功能 |[未使用]
```
Comment (huan) [inline _posts/2018-05-06-talk-about-wechat-robot.md]: github。com -> github.com

File: _posts/2018-05-06-talk-about-wechat-robot.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+### 基于网页版微信 http api lib
+
+这种库就很多了，我们的`wechaty`核心也是这种方式，还有其他各种语言的实现:
+
+* [youfou/wxpy](https://github。com/youfou/wxpy) | 优秀的 api 包装和配套插件，微信机器人/优雅的微信个人号 API|[个人评价: 支持 python2/3 api 包装不错 文档写的满分 基于 itchat 开发，也开放了 itchat 的原始数据接口，自由度和封装美化兼顾 ]
+
+* [liuwons/wxBot](https://github。com/liuwons/wxBot)| 类似的基于 Python 的微信机器人 | [个人评价: 不支持 python3 utf-8 支持不好 ]
+
+* [sjdy521/Mojo-Weixin](https://github。com/sjdy521/Mojo-Weixin)|使用 Perl 语言编写的微信客户端框架，可通过插件提供基于 HTTP 协议的 api 接口供其他语言调用 |[未使用]
```
Comment (huan) [inline _posts/2018-05-06-talk-about-wechat-robot.md]: github。com -> github.com

File: _posts/2018-05-06-talk-about-wechat-robot.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+<!--more-->
+
+
+首先，谢谢[Huan LI](https://github。com/zixia)的邀请，让我有机会写这么一篇博文。其实我做跟微信机器人相关的调研也不是很久，而且中间中断了很长时间，最早是在去年 6 月份开始，做了两个月更换了几个版本和不同方式把产品做了出来，之后就是今年 4 月份，又有这方面的需求于是又重新接触了微信机器人。下面是稍微详细的内容。
+
+### 基于网页版微信 http api lib
+
+这种库就很多了，我们的`wechaty`核心也是这种方式，还有其他各种语言的实现:
+
+* [youfou/wxpy](https://github。com/youfou/wxpy) | 优秀的 api 包装和配套插件，微信机器人/优雅的微信个人号 API|[个人评价: 支持 python2/3 api 包装不错 文档写的满分 基于 itchat 开发，也开放了 itchat 的原始数据接口，自由度和封装美化兼顾 ]
```
Comment (huan) [inline _posts/2018-05-06-talk-about-wechat-robot.md]: github。com -> github.com

File: _posts/2018-05-06-talk-about-wechat-robot.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
+---
+title: "（以我粗鄙的见识）谈谈微信机器人"
+author: nbwsc
+date: '2018-06-05 14:07:52 +0800'
+published: true
+---
+
+> Author: [nbwsc](https://github。com/nbwsc)
```
Comment (huan) [inline _posts/2018-05-06-talk-about-wechat-robot.md]: github。com -> github.com

**Review by huan — CHANGES_REQUESTED** [review]
Please fix the url dot bug, then I could be able to merge the new blog!

Thank you!

File: _posts/2018-05-06-talk-about-wechat-robot.md (hunk 1 above)
Comment (nbwsc) [reply _posts/2018-05-06-talk-about-wechat-robot.md]: sorry, 全局替换的时候忘了这个了...

**Comment by huan** [comment]
You can just keep pushing new commits to your cloned repository, and then this PR will be updated automatically. :)

**Comment by huan** [comment]
@nbwsc Thank you very much for your contribution!

> You've invited Wangshichao to Wechaty! They'll be receiving an email shortly. They can also visit https://github.com/wechaty to accept the invitation.

I have invited you to join our Wechaty GitHub Organization, please accept it by following the above message.

And we also have a WeChat room for contributors only which can discuss Wechaty at a deep level, you are welcome to join and if you are interested please contact @lijiarui to ask for an invitation.

Cheers!

**Pull request merged**